src/support/bits.h

```diff
@@ -65,17 +65,21 @@ template<typename T> int CountTrailingZeroes(T v) {
 template<typename T> int CountLeadingZeroes(T v) {
   return CountLeadingZeroes(typename std::make_unsigned<T>::type(v));
 }
-template<typename T> bool IsPowerOf2(T v) { return v != 0 && PopCount(v) == 1; }
+template<typename T> bool IsPowerOf2(T v) {
+  return v != 0 && (v & (v - 1)) == 0;
+}
 
 template<typename T, typename U> inline static T RotateLeft(T val, U count) {
-  T mask = sizeof(T) * CHAR_BIT - 1;
+  auto value = typename std::make_unsigned<T>::type(val);
```

Comment:
pretty important cast to `unsigned`. Otherwise not LLVM not GCC can't fold this to `rol` / `rot`